### ISSUE TITLE
Traversable rules need inheritDoc when implementing an interface

### ISSUE THREAD

**Issue by spaceemotion** [slevomat/coding-standard#812]
(using v6.0.5)

---

I got the following situation:

- An interface with a method that has an iterable value (array) as parameter/return type, fully documented (e.g. `@return SomeObj[]`)
- An object implementing said interface, without PHPDocs
- The new traversable sniffs enabled

```php
interface Event extends JsonSerializable
{
    /**
     * Recreates the event instance from the serialized data.
     *
     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint
     * @param  array  $payload
     * @return static|Event
     */
    public static function fromJson(array $payload): self;
}

final class Snapshot implements Event
{
    private array $payload;

    /**
     * {@inheritDoc}
     */
    public static function fromJson(array $payload): self
    {
        $instance = new self();
        $instance->payload = $payload;

        return $instance;
    }

    /**
     * {@inheritDoc}
     */
    public function jsonSerialize(): array
    {
        return $this->payload;
    }
}
```

Without the `@inheritDoc` comments I am getting errors about missing `@return`/`@param` annotations. Interestingly enough, even though I have the `UselessInheritDocComment` sniff enabled, it doesn't get triggered.

Is it somehow not inheriting the docblocks properly?

**Comment by kukulich** [comment]
What specific errors do you get?

**Comment by spaceemotion** [comment]
For the source code above, removing the inheritDoc lines results in the following:

```log
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 16 | ERROR | Method
    |       | \xxx\xxx\Snapshot::fromJson()
    |       | does not have @param annotation for its traversable
    |       | parameter $payload.
 24 | ERROR | Method
    |       | \xxx\xxx\Snapshot::jsonSerialize()
    |       | does not have @return annotation for its traversable
    |       | return value.
----------------------------------------------------------------------
```

**Comment by kukulich** [comment]
“Array” is not valid type hint for this sniff. Use “int[]” etc

**Comment by spaceemotion** [comment]
The issue I'm having is not that there're no annotations with the specific type. It's that the inherited method does not inherit the documentation of its parent, if it is empty/missing.

As you can see from `Event#fromJson`, I have the type hints specified there, but it's giving me errors in the "Snapshot" implementation.

**Comment by kukulich** [comment]
PHPCS cannot read inherited annotations.
